platforms/ios/example/Wysiwyg/Views/WysiwygSuggestionList.swift

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright 2023 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+import WysiwygComposer
+
+struct WysiwygSuggestionList: View {
+    @EnvironmentObject private var viewModel: WysiwygComposerViewModel
+    var suggestion: SuggestionPattern
+
+    var body: some View {
+        HStack {
+            VStack(alignment: .leading) {
+                switch suggestion.key {
+                case .at:
+                    let users = Users.filtered(with: suggestion.text)
+                    if !users.isEmpty {
+                        Text(Users.title).underline()
+                        ForEach(users) { user in
+                            Button {
+                                viewModel.setAtMention(link: user.url, name: user.name)
+                            } label: {
+                                HStack(spacing: 4) {
+                                    Image(systemName: user.iconSystemName)
+                                    Text(user.name)
+                                }
+                            }
+                            .accessibilityIdentifier(user.accessibilityIdentifier)
+                        }
+                    }
+                case .hash:
+                    let rooms = Rooms.filtered(with: suggestion.text)
+                    if !rooms.isEmpty {
+                        Text(Rooms.title).underline()
+                        ForEach(rooms) { room in
+                            Button {
+                                viewModel.setHashMention(link: room.url, name: room.name)
+                            } label: {
+                                HStack(spacing: 4) {
+                                    Image(systemName: room.iconSystemName)
+                                    Text(room.name)
+                                }
+                            }
+                            .accessibilityIdentifier(room.accessiblityIdentifier)
+                        }
+                    }
+                case .slash:
+                    let commands = Commands.filtered(with: suggestion.text)
+                    // FIXME: we should probably add the suggestion.start test at a deeper level (Rust?)
+                    if !commands.isEmpty, suggestion.start == 0 {
```

Comment:
Hum, maybe, I'm guessing we could try to move some of the logic in the iOS platform library code, but here apart from the `suggestion.start == 0`, this is just the example app filtering its mock objects. 
I guess it could make sense to provide that kind of view model in the library, and make some protocol to represent a user/room/command and have the client provide objects that could conform to it (e.g. make MXUser/MXMember in Element conform)

Reply:
At least, I moved the /command filtering to Rust as it makes sense on all platforms.

Reply:
I wonder if we should bother with my protocol idea up here ^ 
With the current implementation we are giving freedom to the hosting application to define it's own way of filtering. For example an hosting app could choose that searching for pattern "li" should return "Alice" (i.e. what Element-Web does), but another app could consider that it's mandatory to only allow pattern with matching leading characters.